docs/rfcs/749-unify-path-behavior.md

```diff
@@ -0,0 +1,89 @@
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-749: Unify Path Behavior
+
+Previous Discussion:
+
+- [Specify the behavior of Path](https://forum.beyondstorage.io/t/topic/195)
+- [OS-specific path separator problem](https://github.com/beyondstorage/go-service-ftp/issues/22)
+
+## Background
+
+Unix and Windows all use a different syntax for filesystem paths. Normally, when specifying file paths on Windows, we would use backslashes(`\`). However, in Java, and many other places outside the Windows world, backslashes are the escape character, so we have to double them up, like `C:\\trash\\blah\\blah`. Forward slashes(`/`), on the other hand, do not need to be doubled up and work on both Windows and Unix. There is no harm in having double forward slashes. They do nothing to the path and just take up space (`//` is equivalent to `/./`).
+
+In our services, we have to handle paths like path splicing and path conversion. And here comes the problems:
+
+- Different path separators may exist in a path at the same time, is it possible or how to use cross-platform paths for users?
+- For fs storage service on Windows platform, the behavior of `work_dir` is undefined. Drive letters in connection string cannot be handled for Windows at this stage.
+
+## Proposal
+
+### Absolute Path and Relative Path
+
+All our services should support two kinds of path:
+
+- Absolute Path:
+  - For Unix, the absolute path starts with `/`.
+  - For Windows, the absolute path starts with a drive letter.
+- Relative Path: 
+  - The relative path is for the working directory `WorkDir`.
+  - Path with prefix `./` or `../` is allowed: they are relative paths for file system, but common strings for object storage service.
+
+### WorkDir and path
+
+`WorkDir` specifies the working directory of the process.
+
+- For file system, `WorkDir` SHOULD be an absolute path.
+  - `WorkDir` MUST start with `/` for Unix. For Windows, the passed in `WorkDir` with drive letter SHOULD be allowed.
+  - The default value is `/`, that means the working directory is the root path for Unix, and the current volume of the process for Windows.
+  - Services SHOULD set `WorkDir` to the path name after the evaluation of any symbolic links internal.
+- For object storage, `WorkDir` is the simulated directory or prefix of the object key.
+  - `WorkDir` SHOULD be Unix style with the prefix `/`.
+  - The default value is `/`.
+  - Form service side, the prefix `/` needs to be removed for internal processing, but `/` needs to be added to the fields returned to the user.
+  
+`path` is the file path for file system, or an object key for object storage. Also, it could be a prefix filter for `List` operation.
+
+- For file system services on Windows, `path` MUST be the relative path for `WorkDir`. For file system on Unix, `path` could be an absolute path or a relative path.

```

Comment:
For `go-service-fs` on windows, absolute path is OK which means a path under current volume. But absolute path with driver letter is not allowed.

Reply:
Got it.